include/sharg/validators.hpp

```diff
@@ -583,27 +583,52 @@ class output_file_validator : public file_validator_base
-
-    output_file_validator(output_file_validator const &) = default;             //!< Defaulted.
-    output_file_validator(output_file_validator &&) = default;                  //!< Defaulted.
+    output_file_validator() = default; //!< Defaulted.
+    output_file_validator(output_file_validator const &) = default; //!< Defaulted.
+    output_file_validator(output_file_validator &&) = default; //!< Defaulted.
     output_file_validator & operator=(output_file_validator const &) = default; //!< Defaulted.
-    output_file_validator & operator=(output_file_validator &&) = default;      //!< Defaulted.
+    output_file_validator & operator=(output_file_validator &&) = default; //!< Defaulted.
     virtual ~output_file_validator() = default; //!< Virtual Destructor.
 
     /*!\brief Constructs from a given overwrite mode and a list of valid extensions.
      * \param[in] mode A sharg::output_file_open_options indicating whether the validator throws if a file already
-                       exists.
+     *                 exists.
      * \param[in] extensions The valid extensions to validate for.
      */
-    explicit output_file_validator(output_file_open_options const mode, std::vector<std::string> extensions = {})
-        : file_validator_base{}, mode{mode}
+    explicit output_file_validator(output_file_open_options const mode, std::vector<std::string> const & extensions)
+        : open_mode{mode}
     {
         file_validator_base::extensions_str = detail::to_string(extensions);
         file_validator_base::extensions = std::move(extensions);
     }
 
+    /*!\brief Constructs from a given overwrite mode and a parameter pack of valid extensions.
+     * \param[in] mode A sharg::output_file_open_options indicating whether the validator throws if a file already
+     *                 exists.
+     * \param[in] extensions Parameter pack representing valid extensions. std::string must be constructible from each
+     *                       parameter.
+     */
+    explicit output_file_validator(output_file_open_options const mode, auto && ...extensions)
+        requires ((std::constructible_from<std::string, decltype(extensions)> && ...))
+        : output_file_validator{mode, std::vector<std::string>{std::forward<decltype(extensions)>(extensions)...}}
+    {}
+
+    /*!\brief Constructs from a list of valid extensions.
+     * \param[in] extensions The valid extensions to validate for.
+     */
+    explicit output_file_validator(std::vector<std::string> const & extensions)
+        : output_file_validator{output_file_open_options::create_new, extensions}
+    {}
+
+    /*!\brief Constructs from a parameter pack of valid extensions.
+     * \param[in] extensions Parameter pack representing valid extensions. std::string must be constructible from each
+     *                       parameter.
+     */
+    explicit output_file_validator(auto && ...extensions)
```

Comment:
Since this pack can be empty, this constructor is a valid default constructor. I think that's OK (shouldn't be ambiguous, but not 100% sure). Is there a test for a default-constructed validator?

Reply:
I see that there is a snippet that says "Default constructed validator has an empty extension list." BUT the snippet actually doesn't do default construction. Maybe fix that?

Reply:
The snippet now uses the default ctor.

If there is a default ctor defined, it will use it. 
If it's not defined, it'll use the pack ctor. 
If the default ctor is deleted, it will not use the pack ctor.